webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -427,16 +426,25 @@ const AdvancedTextEditor = ({
         };
     }, [handleDraftChange, draft]);
 
-    // Set the draft from store when changing post or channels, and store the previus one
+    // Keep track of the draft as a ref so that we can save it when changing channels
+    const draftRef = useRef(draftFromStore);
```

Comment:
I flipped the order of these hooks around, moved this ref to where it's actually used, and changed the name of it to make its actual behaviour a bit clearer

Reply:
I like to keep all definitions in the same place, because if not they get lost in between callbacks and effects, and you cannot find anything :P The only exceptions is when you need A to be defined before B (a useMemo requires a useCallback, for example). If we do it like this, instead of knowing "everything is at the beginning of the component" and only need to check in one place, we have to check in two different places (beginning of the component, and before the function). I feel the cognitive load is higher here. Nevertheless, I don't have a strong enough opinion to require to change it back :stuck_out_tongue: 

If you want to keep the whole draft tracking in one single place, you can always move these to a hook. 0/5 though.

Reply:
I generally agree with that, but when the component gets long enough, I think grouping them by purpose makes it easier to find them. It's sort of like how the `utils` and `actions` folders in the app are nice for smaller projects, but they've gotten bloated and hard to navigate now. That's probably just a sign that we should move it to a custom hook though, but while this 100% should be in a hook, I already tried to do that multiple times while working on this, so I don't want to spend more time trying that right now 😅

In the meantime I'll move it back to where it was originally defined to keep the structure consistent with the rest of the file